transformer_engine/pytorch/ops/basic/basic_linear.py

```diff
@@ -553,13 +620,11 @@ def _functional_forward(
             else:
                 torch.distributed.all_reduce(y, group=tensor_parallel_group)
 
-        # Reshape output tensor
-        output_dims = list(input_dims)
-        output_dims[0] = -1
-        output_dims[-1] = weight_dims[0]
-        output = reshape(y, output_dims)
+        # Reshape output tensor if needed
+        if out is None:
+            out = reshape(y, output_dims)
```

Comment:
Shouldn't it be a view instead? I recall that reshape can call memcpy sometimes.

Reply:
This case is only called when there is no user-provided `out`, so reallocation would be valid and unlikely.